app/src/main/java/lv/spkc/apturicovid/ui/intro/SmsViewModel.kt

```diff
@@ -30,13 +31,17 @@ class SmsViewModel @Inject constructor(): BaseViewModel() {
         _smsRetrievalEventLiveData.value = Event(true)
     }
 
-    fun startCodeRetrievalTimer(interval: Long = 1000, showMessage: Boolean = false) {
+    fun startCodeRetrievalTimer(interval: Long = 0L, showMessage: Boolean = false) {
         val timeRemainingMillis = DateTime.now().millis - (lastCodeRequestTime)
-        if (timeRemainingMillis < interval) {
+        if (interval == 0L && lastCodeRequestTime == 0L) {
```

Comment:
I have a feeling that we actually have two big sub-cases here - user can send SMS and she cant. So split the code accordingly :)